graphium/ipu/ipu_losses.py

```diff
@@ -131,14 +176,7 @@ def forward(self, input: Tensor, target: Tensor) -> Tensor:
 
         # Replace the nan-targets in the input/target tensors by 0
         nan_targets = target.isnan()
-        input[nan_targets] = 0.0
-        target[nan_targets] = 0.0
 
         # Compute the loss, and rescale by the number of nan elements
-        loss = super().forward(input, target)
-        num_real_targets = (~nan_targets).sum()
-        factor1 = torch.where(num_real_targets > 0, 1, 0)
-        factor2 = torch.where(num_real_targets > 0, 0, 1)
-        loss = factor1 * loss * nan_targets.numel() / (num_real_targets + factor2)
-
+        loss = super().forward(input, target, nan_targets)
```

Comment:
I think the `nan_targets` are needed inside the forward of the loss function, because first time nans are set to 0s, but after applying softmax, 0s will not stay as 0. This will make the regression loss very large. So we apply `nan_targets` again after softmax.

Reply:
Instead of setting to 0, can you set to -inf? Or to -1000?

Reply:
I think the issue is that unlike masking for the transformer module, we have [0.2, 1.5, 3.2, 0.3, -1000], applying softmax on -1000 would give us 0. In this case it is like [nan, nan, nan] --> [-1000, -10000, -10000] ----> softmax([-1000, -10000, -10000]) --> [0.333, 0.333, 0.333]. It ends up the same as 0s. `F.cross_entropy` takes the `ignore_index` to ignore index -1000. We could also use the `self.weight` parameter like the BCE loss. The l1_loss and mse_loss, on the other hand, does not take `ignore_index` or  `self.weight`.  Maybe we could create a CELossIPU and calculate hybridceloss as: alpha*CELossIPU + (1-alpha)*MSELossIPU